src/kernels/execution/cellExecutionMessageHandler.ts

```diff
@@ -253,21 +279,117 @@ export class CellExecutionMessageHandler implements IDisposable {
     private endCellExecution() {
         this.prompts.forEach((item) => item.dispose());
         this.prompts.clear();
-        this.clearLastUsedStreamOutput();
+        // Assume you have a long running cell, then reload vscode, next this cell continues running.
+        // Next you interrupt this cell, what happens is this will get disposed
+        // as we end up getting a message from for the request kernel info message (generally sent when we re-connect to a kernel)
+        // at that point this method is called and the stream is cleared,
+        // However we still haven't received final reply message
+        // & we could still get one more packet from the kernel, unfortunately the stream has been cleared and we end up appending the new stream
+        // message as a new output , meaning we end up with two outputs instead of one output with multiple output items.
```

Comment:
maybe this large comment should be on the function implementation so that it's not duplicated